### PR TITLE
Add __deserialize__ method to portable atoms

### LINKED ISSUE #32: Subscribe not working on first call for server side portable atom
If a portable atom is created server side and sent back to the client, subscribed functions don't work in quite the same manner.

MWE: https://anvil.works/build#clone:R7UC642A2YGE46CI=LJYJS2572B5QSNXGL5TFAWUI

### PR DESCRIPTION
**Pull request by meatballs**
closes #32

### CODE REVIEW

File: client_code/atomic/atoms.py [hunk 1]
```diff
@@ -114,6 +114,16 @@ def portable_atom(_cls, name=None):
         _cls.__serialize__ = lambda self, _: {
             k: v for k, v in self.__dict__.items() if k != REGISTRAR
         }
+
+    if not any(
+        [hasattr(_cls, attr) for attr in ("__deserialize__", "__new_deserialized__")]
```
Comment (s-cork) [inline client_code/atomic/atoms.py:119]: no need for the `[`?

**Review by s-cork — APPROVED** [review]
one suggestion - lgtm

File: client_code/atomic/atoms.py (hunk 1 above)
Comment (meatballs) [reply client_code/atomic/atoms.py:119]: any needs some form of iterable, so I went with a list comprehension.

Do you have some other preference?

Comment (s-cork) [reply client_code/atomic/atoms.py:119]: Without the square brackets it's just a generator no? Saves us doing two loops. One for the list comp and one for the call to any.

Comment (meatballs) [reply client_code/atomic/atoms.py:119]: sorry if I'm being a bit thick here but I don't quite get what you're saying.

`any` is a function that takes a single iterable argument.
I'm using a list comprehension as that iterable

What am I missing here?

Comment (s-cork) [reply client_code/atomic/atoms.py:119]:
```python
>>> any(i for i in [1,2,3])
True
>>> any(i for i in [0, 0, 0])
False
```

```python
if not any(hasattr(_cls, attr) for attr in ("__deserialize__", "__new_deserialized__")):
   ...
```

Comment (meatballs) [reply client_code/atomic/atoms.py:119]: ahhhh!!! I see. Told you I was being a bit thick!

**Pull request merged**